### PR TITLE
[Soft Delete] Add not null/is null filters to soft delete restore sprocs

### PR DESCRIPTION
**Pull request by cscharf**
Reduce touch/update overhead by filtering on bulk soft-delete and bulk-restore to only impact or consider those rows which are not already in that state.

### CODE REVIEW

File: src/Sql/dbo/Stored Procedures/Cipher_Restore.sql [hunk 1]
```diff
@@ -21,8 +21,9 @@ BEGIN
         [dbo].[UserCipherDetails](@UserId)
     WHERE
         [Edit] = 1
+        AND [DeletedDate] IS NOT NULL
         AND [Id] IN (SELECT * FROM @Ids)
-	
+
```
Comment (kspearrin) [inline src/Sql/dbo/Stored Procedures/Cipher_Restore.sql:26]: That's going on here? Remove whitespace?

**Comment by kspearrin** [comment]
Should we add it to the bulk hard delete one too?

File: src/Sql/dbo/Stored Procedures/Cipher_Restore.sql (hunk 1 above)
Comment (cscharf) [reply src/Sql/dbo/Stored Procedures/Cipher_Restore.sql:26]: Yeah, I have a plug-in that swaps tabs for spaces if it detects mixed tabs/spaces. I'll try to watch erroneous changes like this or just remove that plug-in

**Comment by cscharf** [comment]
> Should we add it to the bulk hard delete one too?

I'm not quite sure... I think there's some value leaving that open for the time being and up to the client to decide... i.e. perhaps the cli automates a hard-delete w/o having to add intermediate steps... maybe the confirmation dialog for delete in the future gets a "permanently delete" checkbox treatment, etc.

File: src/Sql/dbo/Stored Procedures/Cipher_Restore.sql (hunk 1 above)
Comment (kspearrin) [reply src/Sql/dbo/Stored Procedures/Cipher_Restore.sql:26]: Seems like we have extra whitespace indention where not necessary. That's what I was pointing out. We should remove those spaces.

Comment (cscharf) [reply src/Sql/dbo/Stored Procedures/Cipher_Restore.sql]: Fixed

**Pull request merged**